tests/integration/integration_steps/cdf_steps.py

```diff
@@ -0,0 +1,119 @@
+import pandas as pd
+from datetime import datetime
+from time import sleep
+from cognite.client import CogniteClient
+from cognite.client.data_classes import Datapoint, TimeSeries, TimeSeriesWrite
+from cognite.client.exceptions import CogniteNotFoundError
+from cognite.client.data_classes import DataPointSubscriptionWrite
+
+def push_data_points_to_cdf(
+    external_id: str, data_points: list[Datapoint], cognite_client: CogniteClient
+) -> pd.DataFrame:
+    df = pd.DataFrame(columns=["timestamp", "value", "externalId"])
+
+    data_point_list = []
+
+    for datapoint in data_points:
+        data_point_list.append(
+            {
+                "externalId": external_id,
+                "timestamp": str(datapoint.timestamp),
+                "value": datapoint.value
+            }
+        )
+
+    df_original = pd.DataFrame(data_point_list)
+    df = df_original.pivot(index="timestamp", columns="externalId", values="value")
+    df.index = pd.to_datetime(df.index)
+
+    cognite_client.time_series.data.insert_dataframe(df)
+    return df_original
+
+
+def push_time_series_to_cdf(time_series_data: list[TimeSeries], cognite_client: CogniteClient) -> list[TimeSeries]:
+    time_series_write_list = []
+
+    for timeseries in time_series_data:
+        time_series_write_list.append(TimeSeriesWrite(
+            external_id=timeseries.external_id,
+            name=timeseries.name,
+            metadata=timeseries.metadata,
+            is_string=timeseries.is_string,
+            security_categories=timeseries.security_categories,
+            is_step=timeseries.is_step,
+            description=timeseries.description,
+        ))
+
+    cognite_client.time_series.create(time_series_write_list)
+    return time_series_data
+
+def push_data_to_cdf(time_series_data: list[TimeSeries], cognite_client: CogniteClient):
+    time_series_data_points_pushed = {}
+    for ts in time_series_data:
+        time_series_data_points_pushed[ts.external_id] = push_data_points_to_cdf(
+            data_points=ts.datapoints, external_id=ts.external_id, cognite_client=cognite_client
+        )
+    sleep(5)
```

Comment:
Why is that needed?

Reply:
This allows time for the data points to be pushed to CDF before moving on to the next step in the test.  I preferred doing this here than in the test itself

Reply:
is there a more deterministic way to ensure that the push completed before moving on? This might turn into a random, hard to detect test failure whenever the 5 seconds are not enough

Reply:
I had introduced listening to the subscription in our first round of integration testing and that was automatic without  having to wait that may be enough if we want to use that